python/plugins/processing/script/ScriptEdit.py

```diff
@@ -55,3 +55,53 @@ def initShortcuts(self):
                                                            + Qt.Key_Space), self)
         self.shortcutAutocomplete.setContext(Qt.WidgetShortcut)
         self.shortcutAutocomplete.activated.connect(self.autoComplete)
+
+    def toggleComment(self):
```

Comment:
How do you feel about moving this logic to sit in the base QgsCodeEditorPython class instead? Then we could utilise it from everywhere without the duplicate code (even in non-python code parts where QgsCodeEditorPython is used, e.g. in the project macros configuration)

Reply:
That would indeed be better. I will update this PR.